bridge/Mixer.cpp

```diff
@@ -2167,25 +2162,33 @@ void Mixer::engineRecordingStreamRemoved(EngineRecordingStream* engineStream)
 
         // Try first wait for pending jobs without stop the transport
         // as we may have some recording events to be sent and we don't
-        // want to lose them
+        // want to lose them.
+        // Improvement: This could be async as well.
         waitForPendingJobs(200, 5, *transportEntry.second);
         transportEntry.second->stop();
         _engineMixer.getJobManager().abortTimedJobs(transportEntry.second->getId());
 
-        if (!waitForPendingJobs(500, 5, *transportEntry.second))
+        if (transportEntry.second->hasPendingJobs())
         {
-            logger::error("RecordingStream id %s did not finish pending jobs in time. count=%u. Continuing "
-                          "deletion anyway.",
-                _loggableId.c_str(),
-                stream->_id.c_str(),
-                transportEntry.second->getJobCounter().load());
+            initializeOrIncrement(pendingTransportCounter, stream->_endpointIdHash);
+            auto asyncWaitTask = PendingJobsAsyncWaitTask(_loggableId, std::move(transportEntry.second), engineRecordingStream, engineStream->id);
+            asyncWaitTask.setOnTaskEndHandler([this, pendingTransportCounter] () {
```

Comment:
It is a bit unintelligable what is happening when you read this code. Maybe we should ponder a bit if we have made it as simple as it can be done.

Reply:
I can create a a task specific for recording. I avoided that solution because I was trying having the asynWaiter with a template task so I couldn't have a different task type for recording use case.
I changed that and now it can receive any task that extends of AsyncWaitTask so I will create a specialisation for recording and will simplify this